### PR TITLE
Mulebot knockdown nerf

### PR DESCRIPTION
**Pull request by tarlockgod**
## About The Pull Request

Lowers Knockdown from being hit by a hacked MULE from 8 to 4 seconds, knocking people down now also turns the MULE off for 8 seconds becasue it has to reorient itself.

## Why It's Good For The Game

It nerfs the MULEs honestly OP knockdown, and might get a MULE to work with others instead of going on rampages alone.

## Changelog
:cl:
balance: lowerd mule knockdown time, lowered the run over damage from mulebot
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
we shouldnt punish people that use npc mulebots to ride people over, also an 8 second stun for the mule is just a dick move

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -722,9 +722,12 @@
 	if(wires.is_cut(WIRE_AVOIDANCE)) // usually just bumps, but if the avoidance wire is cut, knocks them over.
 		if(iscyborg(L))
 			visible_message("<span class='danger'>[src] bumps into [L]!</span>")
-		else if(L.Knockdown(8 SECONDS))
-			log_combat(src, L, "knocked down")
-			visible_message("<span class='danger'>[src] knocks over [L]!</span>")
+		else
+			L.Knockdown(4 SECONDS)
+			src.turn_off()
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:727]: why would you do src.proc

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 2]
```diff
@@ -722,9 +722,12 @@
...
+			addtimer(CALLBACK(src, .proc/turn_on), 80)
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:728]: use timer defines

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 1 above)
Comment (tarlockgod) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:727]: relativly new to byond code, what can i do instead?

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 2 above)
Comment (tarlockgod) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:728]: Can you give me a example for a timer define? Relativiely new to byond code

**Comment by tarlockgod** [comment]
If someone uses a NPC MULE to knockdown people and then kill them while the MULE is offline, thats fine. But my issue is that some players always pick PAI to use a hacked mule to validhunt. Also, the vanilla 8 second knockdown from the MULE is just absurd.

I could change the MULE downtime to 3 or 5 seconds randomly, if you think thats fine

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:727]:
```suggestion
			turn_off()
```
`src` is implied

File: code/modules/mob/living/simple_animal/bot/mulebot.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm:728]:
```suggestion
			addtimer(CALLBACK(src, .proc/turn_on), 8 SECONDS)
```
This is a timer define. Feel free to ask for help in our coder channel.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 3]
```diff
@@ -722,9 +722,12 @@
 	if(wires.is_cut(WIRE_AVOIDANCE)) // usually just bumps, but if the avoidance wire is cut, knocks them over.
 		if(iscyborg(L))
 			visible_message("<span class='danger'>[src] bumps into [L]!</span>")
-		else if(L.Knockdown(8 SECONDS))
-			log_combat(src, L, "knocked down")
-			visible_message("<span class='danger'>[src] knocks over [L]!</span>")
+		else
+			L.Knockdown(4 SECONDS)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:726]:
```suggestion
		else if(L.Knockdown(4 SECONDS))
```
Keep this check, as some living things can absorb the stun, and then the message would be erroneous.

**Comment by Pugie** [comment]
You could just remove pai functionality.

**Comment by ATH1909** [comment]
> If someone uses a NPC MULE to knockdown people and then kill them while the MULE is offline, thats fine. But my issue is that some players always pick PAI to use a hacked mule to validhunt. Also, the vanilla 8 second knockdown from the MULE is just absurd.
> 
> I could change the MULE downtime to 3 or 5 seconds randomly, if you think thats fine

pAI mulebots can't knock people down just by bumping into them

**Comment by ShibaInuLord** [comment]
Based on what ATH said, this seems like a pointless PR that just removes a functionality from a rare random event and admemeing.
Also, 8 seconds seems way too long

**Comment by davethwave** [comment]
Reminder that the skateboard is the ultimate nemesis to the rouge mulebot as well as other such vehicles and can be easily mass produced.

**Comment by CRITAWAKETS** [comment]
the fact that you nerfed the knockdown (something that barely happens) and not the ludicrous damage of the mulebot just screams of ided

**Comment by FloranOtten** [comment]
> If someone uses a NPC MULE to knockdown people and then kill them while the MULE is offline, thats fine. But my issue is that some players always pick PAI to use a hacked mule to validhunt. Also, the vanilla 8 second knockdown from the MULE is just absurd.

"some players"
One. One player. I could give you his ckey. 

8 second stun for the bot is just stupid. Lower the knockdown, sure, but this makes mulebots actively dogshit.

**Comment by tarlockgod** [comment]
OK, turns out PAIs really cant knock you down, didnt notice that becasue i cant read apparently and i only tested it by assuming control over the MULE. knockdown stays at 4 seconds though, becasue 8 are ridicolous. Also yeah, turning it off is shitty when its a event MULE so that is gonna be removed.

I can however lower the damage when you are being driven over, maybe lower the damage modifiers a bit.

**Comment by Istoprocent** [comment]
Controlled mulebot should have safeties and prevent running over people altogether, unless it has been emagged.

What probably needs to be removed is mulebot's ability to throw bodies, which is the primary method being abused.

**Comment by tarlockgod** [comment]
Realized this PR is kinda dumb and if i wanna do what i want i might need to rewrite so much shit, so this PR is closed now

**Pull request closed without merging**